gk/main.c

```diff
@@ -573,6 +639,12 @@ gk_proc(void *arg)
 
 	gk_conf_hold(gk_conf);
 
+	if (block_idx == 0) {
+		ret = gk_hold_arp_and_nd(gk_conf, lcore);
```

Comment:
Yes, but you shouldn't assume that there's only ONE gateway, there may be a number of gateways.